src/Captioning/Format/SubripFile.php

```diff
@@ -6,9 +6,28 @@
 
 class SubripFile extends File
 {
-    const PATTERN = '#[0-9]+(?:\r\n|\r|\n)([0-9]{2}:[0-9]{2}:[0-9]{2},[0-9]{3}) --> ([0-9]{2}:[0-9]{2}:[0-9]{2},[0-9]{3})(?:\r\n|\r|\n)((?:.*(?:\r\n|\r|\n))*?)(?:\r\n|\r|\n)#';
-
-    protected $lineEnding = File::WINDOWS_LINE_ENDING;
+    const PATTERN =
+        '/^
+                       ### First subtitle ###
+        [\p{C}]{0,3}                            # BOM
+        [\d]+                                   # Subtitle order.
+        ((?:\r\n|\r|\n))                        # Line end.
+        [\d]{2}:[\d]{2}:[\d]{2},[\d]{3}         # Start time.
+        [ ]-->[ ]                               # Time delimiter.
+        [\d]{2}:[\d]{2}:[\d]{2},[\d]{3}         # End time.
+        \1                                      # Line end.
+        (?:[\S ]+\1)+                           # Subtitle text.
+                       ### Other subtitles ###
+        (?:
+            \1(?<=\r\n|\r|\n)[\d]+\1
+            [\d]{2}:[\d]{2}:[\d]{2},[\d]{3}
+            [ ]-->[ ]
+            [\d]{2}:[\d]{2}:[\d]{2},[\d]{3}
+            \1(?:[\S ]+\1)+
+        )*
+        \1?
+        $/xu'
+    ;
```

Comment:
Thank you. I am often use /x modifier in regular expressions when they are too difficult for understanding in one line.